sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -594,5 +600,16 @@ static void _receive(ng_pktsnip_t *pkt)
     ng_ipv6_demux(iface, pkt, hdr->nh);
 }
 
+static void _handle_ipv6_encapsulation(ng_pktsnip_t *pkt)
+{
+    ng_pktsnip_t *ptr = pkt;
+
+    /* Remove encapsulating IPv6 header */
+    while ((ptr->next != NULL) && (ptr->next->type == NG_NETTYPE_IPV6)) {
```

Comment:
That is not what is happening here: its supposed to remove the outer-most IPv6 header + extension headers. But one bug is in there (`pkt->type` was set to `NG_NETTYPE_IPV6` in l73, so it (the encapsulated) is removed here to). Will fix.